crates/turborepo-lib/src/run/cache.rs

```diff
@@ -281,4 +289,65 @@ impl TaskCache {
 
         Ok(cache_status)
     }
+
+    pub async fn save_outputs(
+        &mut self,
+        prefixed_ui: &mut PrefixedUI<impl Write>,
+        duration: Duration,
+    ) -> Result<(), Error> {
+        if self.caching_disabled || self.run_cache.reads_disabled {
+            return Ok(());
+        }
+
+        debug!("caching outputs: outputs: {:?}", &self.repo_relative_globs);
+
+        let files_to_be_cached = globwalk::globwalk(
+            &self.run_cache.repo_root,
+            &self.repo_relative_globs.inclusions,
+            &self.repo_relative_globs.exclusions,
+            globwalk::WalkType::All,
+        )?;
+
+        let mut relative_paths = files_to_be_cached
+            .into_iter()
+            .map(|path| {
+                AnchoredSystemPathBuf::relative_path_between(&self.run_cache.repo_root, &path)
+            })
+            .collect::<Vec<_>>();
+        relative_paths.sort();
+        self.run_cache
+            .cache
+            .put(
+                self.run_cache.repo_root.clone(),
+                self.hash.clone(),
+                relative_paths.clone(),
+                duration.as_millis() as u64,
+            )
+            .await?;
+
+        let notify_result = match self.daemon_client.as_mut() {
+            Some(daemon_client) => daemon_client
+                .notify_outputs_written(
+                    self.hash.to_string(),
+                    self.repo_relative_globs.inclusions.clone(),
+                    self.repo_relative_globs.exclusions.clone(),
+                    duration.as_millis() as u64,
+                )
+                .await
+                .map_err(Error::from),
+            None => Err(Error::NoDaemon),
```

Comment:
I think in the Go implementation we don't error if the daemon doesn't exist. We just return nil. However, this does seem to be the more sensible approach